ext/http/lib.rs

```diff
@@ -312,20 +316,21 @@ fn req_headers(
 }
 
 fn is_websocket_request(req: &hyper::Request<hyper::Body>) -> bool {
-  req_header_contains(req, hyper::header::CONNECTION, "upgrade")
-    && req_header_contains(req, hyper::header::UPGRADE, "websocket")
+  req.version() == hyper::Version::HTTP_11
+    && req.method() == hyper::Method::GET
+    && req.headers().contains_key(&SEC_WEBSOCKET_KEY)
+    && header(req.headers(), &SEC_WEBSOCKET_VERSION) == b"13"
+    && header(req.headers(), &UPGRADE).eq_ignore_ascii_case(b"websocket")
+    && header(req.headers(), &CONNECTION).eq_ignore_ascii_case(b"upgrade")
```

Comment:
wouldn't cause #11494 again?

Reply:
Oh, so some clients do send multiple tokens... Interesting, because my Firefox doesn't. Anyway, I'll fix it.

Reply:
Unsettling. There's a httpUpgradeWebSocketMultipleConnectionOptions test that exercises multiple connection header tokens... and it passes.

Reply:
That test is for JS only, it doesnt rely on the code you changed